### DISCUSSION TITLE
Remove some columns but make it available via "Add column..."

### DISCUSSION THREAD

**Discussion by rsakai** [lineupjs/lineupjs#485] [Q&A] [answered]
Is it possible to remove some columns, but have them available under the "Add column..." dropdown?
![image](https://user-images.githubusercontent.com/554331/138847470-7e363b8d-a1a7-43c9-9ce4-fd7beaffe640.png)

Another way to ask this is, is it possible to trigger the "remove"  on the LineUp object?
![image](https://user-images.githubusercontent.com/554331/138847275-380b5c40-ae2f-4fb0-9277-a4f5fde5abdd.png)

I have too many columns and I want to start with a basic set and give the user the option to add the columns they want.  Maybe there is a way to do this from the builder, but I can see only those columns that I added via column()  function under the "Add column..." dropdown.   Any work around? or did I miss something obvious?

**Comment by sgratzl** [comment, marked as answer]
when you create lineup using the builder you can define how the ranking is being built. 

e.g. in https://github.com/lineupjs/lineupjs/blob/6a6a7244b87ed33538d4efccd727efe933ba998d/demo/builder2.html#L36-L41

it creates a custom ranking with a specific sorting and a defined set of columns. I assume in your case only the "defaultRanking" is created which adds all available columns by default.

**Reply by rsakai** [reply]
Thanks @sgratzl.  I did something like the below and it is working.  I hope this is what you meant...

```js
 // create a builder
    const builder = LineUpJS.builder(my_data);
    // derive all columns
    builder.deriveColumns().deriveColors();
    // create a ranking
    const ranking = LineUpJS.buildRanking()
      .supportTypes()
      .column("col_01")
      .column("col_02")
      .column("col_03");
   // set ranking
    builder.ranking(ranking);
    const lineup = builder.build(document.getElementById("my-chart"));
```
